packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/tile.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'package:meta/meta.dart' show immutable;
+
+/// Contains information about a Tile that is returned by a [TileProvider].
+@immutable
+class Tile {
+  /// Creates an immutable representation of a [Tile] to draw by [TileProvider].
+  const Tile(this.width, this.height, this.data);
+
+  /// The width of the image encoded by data in pixels.
+  final int width;
+
+  /// The height of the image encoded by data in pixels.
+  final int height;
+
+  /// A byte array containing the image data.
```

Comment:
What's the format?

Reply:
By looking at the doc, it can be any image data format
https://developers.google.com/maps/documentation/android-sdk/reference/com/google/android/libraries/maps/model/Tile#public-final-byte[]-data
Maybe we can add an example on how to get the data?
Like 
```
/// One way to generate the data is to use the [toByteData](https://api.flutter.dev/flutter/dart-ui/Image/toByteData.html) method. For example:
/// ```
/// ByteData byteData = await image.toByteData(format: ui.ImageByteFormat.png);
/// Uint8List byteData =  byteData.buffer.asUint8List();
/// ```
```

Reply:
hmm. Would adding `The image data format accepted by platforms various based on platforms and versions.` enough?

Reply:
Maybe:

The image data format must be one that's natively supported for decoding by the platform (e.g on Android it can only be one of the [supported image formats for decoding](https://developer.android.com/guide/topics/media/media-formats#image-formats).

Reply:
done